server/protocols/webadmin/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java

```diff
@@ -0,0 +1,135 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.exception.MailboxExistsException;
+import org.apache.james.mailbox.exception.MailboxNotFoundException;
+import org.apache.james.mailbox.model.MailboxMetaData;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MailboxQuery;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.util.streams.ImmutableCollectors;
+import org.apache.james.webadmin.model.MailboxResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+
+public class UserMailboxesService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserMailboxesService.class);
+    private static final String USER_NAME = "webAdmin";
+
+    private final MailboxManager mailboxManager;
+    private final UsersRepository usersRepository;
+
+    @Inject
+    public UserMailboxesService(MailboxManager mailboxManager, UsersRepository usersRepository) {
+        this.mailboxManager = mailboxManager;
+        this.usersRepository = usersRepository;
+    }
+
+    public void createMailbox(String username, String mailboxName) throws MailboxException, UsersRepositoryException {
+        usernamePreconditions(username);
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(mailboxName));
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(USER_NAME, LOGGER);
+        try {
+            mailboxManager.createMailbox(
+                convertToMailboxPath(username, mailboxName, mailboxSession),
+                mailboxSession);
+        } catch (MailboxExistsException e) {
+            LOGGER.info("Attempt to create mailbox {} for user {} that already exists", mailboxName, username);
+        }
+    }
+
+    public void deleteMailboxes(String username) throws MailboxException, UsersRepositoryException {
```

Comment:
Do you mean I should be deleting children mailboxes ?
What do you think about not deleting mailboxes that have children ? (As it can be dangerous to assume so)

Note : here we want to delete all the mailboxes of the user. It is intended for other use cases than deleting a single mailbox.